### PR TITLE
Contentful v5

### PR DESCRIPTION
**Pull request by nmoskaleva**

### CODE REVIEW

File: contentful.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+// Features
+const contactInfoEntryId = '57BmNlPMn5pBZBBvPNIoLC'
+
+// FETCH ENTRIES
+// by entry ID
+async function fetchEntry(entryId) {
+    const entry = await client.getEntry(entryId)
+    const entryData = entry.fields
+    if (entry.fields) return entryData
+    else console.log(`error fetching content`)
```
Comment (benna100) [inline contentful.js:40]:
```suggestion
    const {fields}= await client.getEntry(entryId)
    if (fields) return fields;
    console.log(`error fetching content`)
```

File: contentful.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+// by content type
+async function fetchEntries(contentType) {
+    const entries = await client.getEntries({
+        content_type: contentType,
+        order: 'fields.id'
+    })
+    const entriesList = entries.items
+    if (entries.items) return entriesList
+    else console.log(`error fetching data`)
```
Comment (benna100) [inline contentful.js:51]: Same as above

File: contentful.js [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+}
+
+// PAGES
+// HOME Page
+export async function fetchHomePageContent() {
+    const entry = await fetchEntry(homePageEntryId)
+    const title = entry.headline
+    const content = entry.mainBody
+    return { title, content }
+}
```
Comment (benna100) [inline contentful.js:61]: We need a wrapper function for this. Maybe one we use like this:

`const homepage = getpageContent('home');`

File: contentful.js [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+// ABOUT Page
+export async function fetchAboutContent() {
+    const entry = await fetchEntry(aboutPageEntryId)
+    const title = entry.headline
+    const content = entry.mainBody
+    return { title, content }
+}
```
Comment (benna100) [inline contentful.js:69]: Why do you want to change the keys? why not just return entry?

File: pages/apply.js [hunk 5]
```diff
@@ -34,7 +40,22 @@ export default () => (
...
+  const requirements = await fetchRequirements()
+
+  return {
+    props: {
+      title: pageContent.title, contentOne: pageContent.contentOne, contentTwo: pageContent.contentTwo,
+      requirementsList: [...requirements],
+      faq: [...faq]
+    }
+  }
+}
```
Comment (benna100) [inline pages/apply.js:61]: Okay now i see what you mean. So you fetch the data for each page and that data can then be passed down to its components. Hmm

Its really really good work Natalia but im thinking from a developer perpespective how much work goes into adding a new component:

1. I must first create the function to get the data in the contentful.js and export this function.
2. I need to add every piece of content i need to the getStaticProps object, add the correct parameter to page function, and then feed the right content into the relevant component.
Compared to importing a json file and using the correct key. 

I think there is too many steps here. The purpose of adding contentful should make it easier to add content. 

But i think as you correctly wrote that if we fetch all the data with the `npm run export-content` that will make it easier. Would that be a lot of work?

File: pages/volunteer.js [hunk 6]
```diff
@@ -6,17 +6,22 @@ import { MentorsTeam } from '../components/team/team'
...
+
+export default ({ mentors_title, mentors_list, partners, sponsor }) => (
   <Layout>
     <Head>
       <title>{title}</title>
     </Head>
     <Content>
       <div>
-        {documentToReactComponents(content)}
+        {documentToReactComponents(content[0])}
```
Comment (benna100) [inline pages/volunteer.js:24]: Why do we need to do this? Cant we do just like before where we had a local json file we load in. Like in the faq example?

Comment (nmoskaleva) [reply pages/volunteer.js:24]: Unfortunately, the structure of JSON that comes from Contentful doesn't allow it:( `content` here is a rich text, and has to be rendered with `documentToReactComponents`. 
Even when the content is not a rich text, the structure is quite complex (like in `faq-mentors.js`)

**Pull request closed without merging**